sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -835,22 +835,23 @@ case class DescribeColumnCommand(
 case class ShowTablesCommand(
     databaseName: Option[String],
     tableIdentifierPattern: Option[String],
+    attributes: Seq[Attribute] = Seq.empty,
     isExtended: Boolean = false,
     partitionSpec: Option[TablePartitionSpec] = None) extends RunnableCommand {
 
-  // The result of SHOW TABLES/SHOW TABLE has three basic columns: database, tableName and
-  // isTemporary. If `isExtended` is true, append column `information` to the output columns.
-  override val output: Seq[Attribute] = {
-    val tableExtendedInfo = if (isExtended) {
-      AttributeReference("information", StringType, nullable = false)() :: Nil
-    } else {
-      Nil
-    }
-    AttributeReference("database", StringType, nullable = false)() ::
+  override val output: Seq[Attribute] = if (attributes.isEmpty) {
+    // The output should be delivered by logical nodes(ShowTables and ShowTableExtended) in normal.
+    // Since the output is not specified when constructing ShowTablesCommand in SQLContext.tables,
```

Comment:
if `SQLContext.tables` is the only exception, can we put this special logic there?

Reply:
OK